### DISCUSSION TITLE
Integration of control tower and gruntwork reference architecture

### DISCUSSION THREAD

**Discussion by velarx** [gruntwork-io/knowledge-base#507] [Help]
I am having few queries related to the gruntwork reference architecture integration with our existing aws cloud infrastructure. Currently we are already running with few account like network, logs, shared, platform  in aws under root account and we wanted to get setup the reference architecture. So what would be best practice to get it setup with existing cloud?

[r:terraform-aws-architecture-catalog](https://github.com/gruntwork-io/terraform-aws-architecture-catalog)
https://github.com/gruntwork-io/terraform-aws-cis-service-catalog

---

<ins datetime="2022-07-15T09:09:35Z">
  <p><a href="https://support.gruntwork.io/hc/requests/108989">Tracked in ticket #108989</a></p>
</ins>

**Comment by pete0emerson** [comment]
Our reference architecture requires newly provisioned AWS accounts.

There are a few migration patterns that we see for customers migrating their existing infrastructure over to the Reference Architecture. Generally, these patterns would be done on a per-environment basis, starting with your least critical environments and working your way up to production.

**Blue / Green**

1) Get your infrastructure up and running in the new Reference Architecture parallel to production
2) Test it with mock data (the more production-like the better)
3) Put your persistent data into read-only mode
4) Copy your persistent data over to the Reference Architecture
5) Test
6) Re-point DNS at the Reference Architecture
7) Test
8) ???
9) Profit!

**Piecemeal**

1. Connect the Reference Architecture to your existing architecture (VPC peering, et cetera)
2. Do the above steps 1-9 but on a much smaller scale
3. Shut down the old architecture once all migrations are complete

Both of these patterns have their pros and cons. Choosing between them will specific to your company and your architecture. Non-production environments should be easier to move than production due to less of a concern about data loss and customer optics.

Note that the task of migrating your infrastructure to the Reference Architecture is up to you; Gruntwork currently doesn't have services to do that migration for you. However, we do work with a network of consultants that do this type of work, and we'd be very happy to connect you if that is a path you'd like to consider.

**Reply by velarx** [reply]
Thank you @pete0emerson.

Scenario 1:
Well we try to go with the new aws account along with the reference architecture then In that case how we are going to manage the account for the multiple product. In current scenario gruntwork is having shared, logs, security, dev, stage and prod to the root account. Well in my case I would be having 300 applications that is running with the dev, stage and prod how we are going to manage the aws account with respect to reference architecture in that case for other products.
Please consider the billing point of view also with respect to those application. Also is there any way we can use the reference architecture along with the aws control tower for those application which we need to migrate to the cloud?

Scenario 2:
What if I want to setup reference architecture at child node with existing aws cloud? If it is possible the what are the ways we can achieve with scenario 1 and combining all the shared, logs and security with root level according to CIS compliance. For example:

```
- root
        =>non gruntwork managed shared
        =>non gruntwork managed logs
        =>non gruntwork managed security
        =>non gruntwork managed network
        =>gruntwork managed platform
                          >gruntwork managed shared
                          >gruntwork managed logs
                          >gruntwork managed security
                          >gruntwork managed dev
                          >gruntwork managed stage
                          >gruntwork managed prod
```

**Reply by deepaucksharma** [reply]
@pete0emerson  Thanks for your reply.

We are trying to figure out how gruntwork reference architecture can co-exist along with an existing landing zone that was set up using AWS Control Tower.  This is illustrated by @icfai-aniketsingh  in scenario 2.
